metrics/testing/metrictest.cpp

```diff
@@ -0,0 +1,171 @@
+#include <cstdio>
```

Comment:
This file should be cleaned up so it only contains tests for the current code.  It should also probably integrate into the cppunit unit test framework - but that can be done as a different pr/milestone.